datadog/resource_datadog_user.go

```diff
@@ -90,75 +121,186 @@ func buildDatadogUserStruct(d *schema.ResourceData) *datadog.User {
+	}
 
 	// Datadog does not actually delete users, so CreateUser might return a 409.
 	// We ignore that case and proceed, likely re-enabling the user.
-	if _, err := client.CreateUser(u.Handle, u.Name); err != nil {
-		if !strings.Contains(err.Error(), "API error 409 Conflict") {
+	createResponse, httpresp, err := datadogClientV2.UsersApi.CreateUser(authV2).Body(*userRequest).Execute()
+	if err != nil {
+		if httpresp == nil || httpresp.StatusCode != 409 {
 			return translateClientError(err, "error creating user")
 		}
-		log.Printf("[INFO] Updating existing Datadog user %s", *u.Handle)
-	}
+		email := d.Get("email").(string)
+		log.Printf("[INFO] Updating existing Datadog user %s", email)
+		// Find user ID by listing user and filtering by email
+		listResponse, _, err := datadogClientV2.UsersApi.ListUsers(authV2).Filter(email).Execute()
+		if err != nil {
+			return translateClientError(err, "error searching user")
+		}
+		responseData := listResponse.GetData()
+		if len(responseData) != 1 {
+			return fmt.Errorf("could not find single user with email %s", email)
+		}
+		userId := responseData[0].GetId()
+		userRequest := buildDatadogUserV2UpdateStruct(d, userId)
 
-	if err := client.UpdateUser(*u); err != nil {
-		return translateClientError(err, "error updating user")
+		res, _, err := datadogClientV2.UsersApi.UpdateUser(authV2, userId).Body(*userRequest).Execute()
+		if err != nil {
+			return translateClientError(err, "error updating user")
+		}
+		userData := res.GetData()
+		d.SetId(userData.GetId())
+	} else {
+		userData := createResponse.GetData()
+		d.SetId(userData.GetId())
 	}
 
-	d.SetId(u.GetHandle())
-
 	return resourceDatadogUserRead(d, meta)
 }
 
 func resourceDatadogUserRead(d *schema.ResourceData, meta interface{}) error {
 	providerConf := meta.(*ProviderConfiguration)
-	client := providerConf.CommunityClient
 
-	u, err := client.GetUser(d.Id())
-	if err != nil {
-		return err
-	}
+	if isV2User(d.Id()) {
+		datadogClientV2 := providerConf.DatadogClientV2
+		authV2 := providerConf.AuthV2
+
+		userResponse, _, err := datadogClientV2.UsersApi.GetUser(authV2, d.Id()).Execute()
+		if err != nil {
+			return translateClientError(err, "error getting user")
+		}
 
-	d.Set("disabled", u.GetDisabled())
-	d.Set("email", u.GetEmail())
-	d.Set("handle", u.GetHandle())
-	d.Set("name", u.GetName())
-	d.Set("verified", u.GetVerified())
-	d.Set("access_role", u.GetAccessRole())
-	d.Set("is_admin", u.GetIsAdmin())
+		userData := userResponse.GetData()
+		userAttributes := userData.GetAttributes()
+		d.Set("email", userAttributes.GetEmail())
+		d.Set("name", userAttributes.GetName())
+		d.Set("title", userAttributes.GetTitle())
+		d.Set("verified", userAttributes.GetVerified())
+		d.Set("disabled", userAttributes.GetDisabled())
+		d.Set("handle", userAttributes.GetHandle())
+	} else {
+		client := providerConf.CommunityClient
+		u, err := client.GetUser(d.Id())
+		if err != nil {
+			return err
+		}
+
+		d.Set("email", u.GetEmail())
```

Comment:
`disabled` should still be set in v1 too no ?

Reply:
Yeah removed it by mistake.